content/docs/rfcs/27/README.md

```diff
@@ -0,0 +1,98 @@
+---
+slug: 27
+title: 27/WAKU2-PEERS
+name: Waku v2 Client Peer Management Recommendations
+status: raw
+editor: Hanno Cornelius <hanno@status.im>
+contributors:
+---
+
+`27/WAKU2-PEERS` describes a recommended minimal set of peer storage and peer management features to be implemented by Waku v2 clients.
+
+In this context, peer _storage_ refers to a client's ability to keep track of discovered or statically-configured peers and their metadata.
+It also deals with matters of peer _persistence_,
+or the ability to store peer data on disk to resume state after a client restart.
+
+Peer _management_ is a closely related concept and refers to the set of actions a client MAY choose to perform based on its knowledge of its connected peers,
+e.g. triggering reconnects/disconnects, keeping certain connections alive, etc.
+
+# Peer store
+
+The peer store SHOULD be an in-memory data structure where information about discovered or configured peers are stored.
+It SHOULD be considered the central authority for peer-related information in a Waku v2 client.
+Clients MAY choose to persist this store on-disk.
+
+## Tracked peer metadata
+
+It is RECOMMENDED that a Waku v2 client tracks at least the following information about each of its peers in a peer store:
+
+| Metadata | Description  |
+| --- | --- |
+| _Public key_  | The public key for this peer. This is related to the libp2p [`Peer ID`](https://docs.libp2p.io/concepts/peer-id/). |
+| _Addresses_ | Known transport layer [`multiaddrs`](https://docs.libp2p.io/concepts/addressing/) for this peer. |
```

Comment:
Do Addresses and Protocols map to the [`storedInfo`](https://github.com/status-im/nim-waku/blob/cd703c6c52bf480213e63413d449879111c1cb35/waku/v2/node/storage/peer/waku_peer_storage.nim#L68) column in the peer table?

Reply:
They do! `storedInfo` is supposed to be a summary view of the client's base libp2p info (`multiaddrs`, `protos`, `key`). Wrapping these in a protobuf makes it easier to serialise and load into a BLOB column in the table.